services/edit/db.js

```diff
@@ -176,13 +176,15 @@ function expectJSONResult(res) {
  */
 function expectHTMLResult(uri) {
   return function (res) {
-    return res.text().then(function (body) {
-      // string -> elements
-      return dom.create(body);
-    })
+    return res.text().then(dom.create) // string -> elements
     .then(function (html) {
-      // add uri
-      html.setAttribute(references.referenceAttribute, uri);
+      if (html.nodeType === 1) {
```

Comment:
Could you use `Node type constants` instead of integers? https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType#Node_type_constants